server/src/main/java/io/crate/metadata/Scalar.java

```diff
@@ -228,6 +228,10 @@ public enum Feature {
         /**
          * If this feature is set, the function will never return null.
          */
-        NON_NULLABLE
+        NON_NULLABLE,
+        /**
+         * If this feature is set, the function may return null for non-null inputs.
+         */
+        NEITHER_NULLABLE_NOR_NON_NULLABLE
```

Comment:
The only usage of this nullability feature is when we generate Lucene queries, where we can decide whether to pre-filter all null elements.
`NEITHER_NULLABLE_NOR_NON_NULLABLE` would be a default tag that prevent such optimizations. Another name could be `unclassified` or `default`.